ml/ml.cc

```diff
@@ -387,20 +397,275 @@ ml_dimension_calculated_numbers(ml_dimension_t *dim, const ml_training_request_t
     }
 
     // Find first non-NaN value.
-    for (idx = 0; std::isnan(tls_data.training_cns[idx]); idx++, training_response.total_values--) { }
+    for (idx = 0; std::isnan(training_thread->training_cns[idx]); idx++, training_response.total_values--) { }
 
     // Overwrite NaN values.
     if (idx != 0)
-        memmove(tls_data.training_cns, &tls_data.training_cns[idx], sizeof(calculated_number_t) * training_response.total_values);
+        memmove(training_thread->training_cns, &training_thread->training_cns[idx], sizeof(calculated_number_t) * training_response.total_values);
 
     training_response.result = TRAINING_RESULT_OK;
-    return { tls_data.training_cns, training_response };
+    return { training_thread->training_cns, training_response };
+}
+
+const char *db_models_create_table =
+    "CREATE TABLE IF NOT EXISTS models("
+    "    dim_id BLOB, dim_str TEXT, after INT, before INT,"
+    "    min_dist REAL, max_dist REAL,"
+    "    c00 REAL, c01 REAL, c02 REAL, c03 REAL, c04 REAL, c05 REAL,"
```

Comment:
@vkalintiris does this mean we'd be hard coding length of cluster centers and number of clusters?

Reply:
Yes, these were already hard coded. We could drop and recreate models if we ever decide to do things differently.

Reply:
yeah cool - think we might consider cleaning up the conf to retire these params maybe or something to make it clear whats hard coded and not - i will think and do in follow on PR as needed.

Reply:
i think it relates to this param only:

```
# num samples to lag = 5
```

Reply:
Hello @vkalintiris ,

Is it possible these values to be `NULL`? If it is not possible, please, set the constraint. 

Best regards!